lib/querytables.js

```diff
@@ -120,42 +120,38 @@ function encodePGName (name) {
 
 /**
  * This is based on the cartodb-postgresql function `CDB_Get_Foreign_Updated_At`
- * @param {Object}   pg         Database connection (PSQL)
- * @param {Object}   table      {id_name, reloid }
- * @param {Function} callback
+ *
+ * @param {Object} pg database connection (PSQL)
+ * @param {Object} table {id_name, reloid }
+ * @returns {Promise<Date>}
+ * @throws {Error}
  */
-function getForeignTableUpdatedAt (pg, table, callback) {
-    const remoteNameQuery = `
-        WITH cdb_fdw_option_row AS
-        (
-            SELECT ftoptions FROM pg_foreign_table WHERE ftrelid='${encodePGName(table.id_name)}'::regclass LIMIT 1
-        ),
-        cdb_fdw_option_table AS
-        (
-            SELECT (pg_options_to_table(ftoptions)).* FROM cdb_fdw_option_row
-        )
-        SELECT
-            FORMAT('%I.%I',
-                (SELECT option_value FROM cdb_fdw_option_table WHERE option_name='schema_name'),
-                (SELECT option_value FROM cdb_fdw_option_table WHERE option_name='table_name')
-            ) AS cdb_fdw_qual_name
+async function getForeignTableUpdatedAt (pg, table) {
+    try {
```

Comment:
This try-catch is catching more things than what it used to, for example is catching the error from `foreignQualifiedName` just to rethrow it with a longer message.

Reply:
And more explicit with the underlying error. I cannot see a problem with that. I can reduce the scope of the try/catch but I'd keep the check for `foreignQualifiedName`.

Reply:
e.g: `Could not fetch metadata for relation 'wadus': Cannot find the names of the foreign tables`